### PR TITLE
Add a way to bulk exclude/include filtered artifacts

### LINKED ISSUE #1076: Allow for fast way to bulk exclude filtered artifacts per character
https://discord.com/channels/785153694478893126/1084490038219382784

Now that excluded are per character rather than global it has removed the feature of bulk management for including/excluding artifacts.
For example you could filter for all Feathers and bulk exclude them. Then include just 1 of them to effectively force the Optimizer to generate the best build around that Feather.

I suggest putting the bulk management buttons; Exclude, Include, Lock, Unlock on the "Select artifact to add to this character's exclusion list" popup window for more power to the user:

image

image

### PR DESCRIPTION
**Pull request by SirDerpyHerp**
## Describe your changes

Added a way to build exclude/include filtered artifacts.
Added pagination for easier navigation

## Issue or discord link

- Resolves #1076 

## Testing/validation

image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code, in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] Ran `yarn run mini-ci` locally to validate format + lint.
- [ ] If there were format issues, I ran `nx format write` to resolve them automatically.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Comment by SirDerpyHerp** [comment]
Side note but, I'm not sure if [1da5aaa](https://github.com/frzyc/genshin-optimizer/pull/1457/commits/1da5aaa156ad6589c4e3395d8271b0590489a752) is a good way of doing things?

**Comment by nguyentvan7** [comment]
https://genshin-optimizer-prs.github.io/pr/1457/frontend

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx [hunk 1]
```diff
@@ -194,6 +202,144 @@ export default function ExcludeArt({
...
+      </Grid>
+      <Grid item xs={12} sm={6} md={3}>
+        <Button
+          fullWidth
+          color="error"
+          disabled={!numExcludedArt}
+          onClick={includeArtifacts}
+          startIcon={<ShowChartIcon />}
+        >
+          <Trans t={t} i18nKey="button.includeArtifacts">
```
Comment (nguyentvan7) [inline apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx:320]: Now that we can include artifacts in this modal, the initial button name doesn't make a lot of sense.
image
@PolomoPT any ideas?

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]
Functionality looks good

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx [hunk 2]
```diff
@@ -273,23 +453,88 @@ function ArtifactSelectModal({
...
+                {artifactIdsToShow.map((id) => (
+                  <Grid
+                    item
+                    key={id}
+                    xs={1}
+                    sx={{ ...(artExclusion.includes(id) && { opacity: 0.4 }) }}
+                  >
+                    <ArtifactCard
+                      artifactId={id}
+                      onClick={clickHandler}
```
Comment (nguyentvan7) [inline apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx:497]: We should change this clickHandler to now include artifacts, if you click an excluded one

Comment (SirDerpyHerp) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx:505]: Implemented

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx [hunk 3]
```diff
@@ -194,6 +202,144 @@ export default function ExcludeArt({
   )
 }
 
+export function ExcludeArtRedButtons({
```
[the remark this answers is not in the document]
Comment (SirDerpyHerp) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx]: Added

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx (hunk 1 above)
Comment (SirDerpyHerp) [reply apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx:328]: Oh related to translations, there's currently no text entry for including artifacts.

**Comment by nguyentvan7** [comment]
Closing due to stale. Feel free to reopen once the branch is up to date and gates are passing.

**Pull request closed without merging**